options/configurable_helper.h

```diff
@@ -75,6 +75,43 @@ class ConfigurableHelper {
       const std::unordered_map<std::string, std::string>& options,
       std::unordered_map<std::string, std::string>* unused);
 
+  // Helper method for configuring a new customizable object.
+  // If base_opts are set, this is the "default" options to use for the new
+  // object.  Then any values in "new_opts" are applied to the object.
+  // Returns OK if the object could be successfully configured
+  // @return NotFound If any of the names in the base or new opts were not valid
+  //      for this object.
+  // @return NotSupported  If any of the names are valid but the object does
+  //       not know how to convert the value.  This can happen if, for example,
+  //       there is some nested Configurable that cannot be created.
+  // @return InvalidArgument If any of the values cannot be successfully
+  //       parsed.
+  static Status ConfigureNewObject(
+      const ConfigOptions& config_options, Configurable* object,
+      const std::string& id, const std::string& base_opts,
+      const std::unordered_map<std::string, std::string>& new_opts);
+
+  // Splits the input opt_value into the ID field and the remaining options.
+  // The input opt_value can be in the form of "name" or "name=value
+  // [;name=value]". The first form uses the "name" as an id with no options The
+  // latter form converts the input into a map of name=value pairs and sets "id"
+  // to the "id" value from the map.
+  // @param opt_value The value to split into id and options
+  // @param id The id field from the opt_value
+  // @param options The remaining name/value pairs from the opt_value
+  // @param default_id If specified and there is no id field in the map, this
+  // value is returned as the ID
+  // @return OK if the value was converted to a map succesfully and an ID was
+  // found.
+  // @return InvalidArgument if the value could not be converted to a map or
+  // there was or there is no id property in the map.
+  static Status GetOptionsMap(
+      const std::string& opt_value, std::string* id,
+      std::unordered_map<std::string, std::string>* options);
+  static Status GetOptionsMap(
+      const std::string& opt_value, const std::string& default_id,
```

Comment:
I did not see any test or examples that we need to use the defualt_id. Do we need this interface? Can we just use Status GetOptionsMap(
      const std::string& opt_value, std::string* id,
      std::unordered_map<std::string, std::string>* options); and in the function, we set id to ""?